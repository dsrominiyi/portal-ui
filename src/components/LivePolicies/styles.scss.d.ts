export const numberOfPolicies: string;
export const policyCount: string;
export const policyText: string;
