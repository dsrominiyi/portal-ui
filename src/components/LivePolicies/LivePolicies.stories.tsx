import React from 'react';
import { storiesOf } from '@storybook/react';
import LivePolicies from '.';

import mockData from '../../../__mocks__/mock-data.json';

const stories = storiesOf('LivePolicies', module);

const render = (policies: Policy[]) => () => <LivePolicies policies={policies} />;

export const storyRenders = {
  withPolicies: render(mockData.policies),
  withNoPolicies: render([])
};

stories.add('With Policies', storyRenders.withPolicies);
stories.add('With No Policies', storyRenders.withNoPolicies);
