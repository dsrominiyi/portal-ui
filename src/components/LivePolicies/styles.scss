@import '../../style/vars';

.number-of-policies {
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: center;
  color: $brand-charcoal;
}

.policy-count,
.policy-text {
  display: flex;
  justify-content: center;
  padding: 5px;
}

.policy-count {
  font-weight: 500;
  font-size: 2rem;
}
