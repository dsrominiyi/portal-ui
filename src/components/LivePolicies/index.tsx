import React, { FunctionComponent } from 'react';

import styles from './styles.scss';

interface Props {
  policies: Policy[];
}

export const LivePolicies: FunctionComponent<Props> = ({ policies }: Props) => (
  <div className={styles.numberOfPolicies}>
    <div className={styles.policyCount}>{policies.length}</div>
    <div className={styles.policyText}>Live Policies</div>
  </div>
);

LivePolicies.displayName = 'LivePolicies';
export default LivePolicies;
