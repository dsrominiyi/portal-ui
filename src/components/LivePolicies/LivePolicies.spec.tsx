import { storyRenders } from './LivePolicies.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('LivePolicies', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
