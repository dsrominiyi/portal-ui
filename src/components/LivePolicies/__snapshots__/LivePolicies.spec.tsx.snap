// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LivePolicies component story renders match snapshots: withNoPolicies 1`] = `
<div
  className="numberOfPolicies"
>
  <div
    className="policyCount"
  >
    0
  </div>
  <div
    className="policyText"
  >
    Live Policies
  </div>
</div>
`;

exports[`LivePolicies component story renders match snapshots: withPolicies 1`] = `
<div
  className="numberOfPolicies"
>
  <div
    className="policyCount"
  >
    3
  </div>
  <div
    className="policyText"
  >
    Live Policies
  </div>
</div>
`;
